src/Symfony/Component/Validator/Tests/Constraints/HostnameValidatorTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\Hostname;
+use Symfony\Component\Validator\Constraints\HostnameValidator;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+/**
+ * @author Dmitrii Poddubnyi <dpoddubny@gmail.com>
+ */
+class HostnameValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator()
+    {
+        return new HostnameValidator();
+    }
+
+    public function testNullIsValid()
+    {
+        $this->validator->validate(null, new Hostname());
+
+        $this->assertNoViolation();
+    }
+
+    public function testEmptyStringIsValid()
+    {
+        $this->validator->validate('', new Hostname());
+
+        $this->assertNoViolation();
+    }
+
+    public function testExpectsStringCompatibleType()
+    {
+        $this->expectException(\Symfony\Component\Validator\Exception\UnexpectedValueException::class);
+
+        $this->validator->validate(new \stdClass(), new Hostname());
+    }
+
+    /**
+     * @dataProvider getValidTldDomains
+     */
+    public function testValidTldDomainsPassValidationIfTldRequired($domain)
+    {
+        $this->validator->validate($domain, new Hostname());
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getValidTldDomains
+     */
+    public function testValidTldDomainsPassValidationIfTldNotRequired($domain)
+    {
+        $this->validator->validate($domain, new Hostname(['requireTld' => false]));
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidTldDomains()
+    {
+        return [
+            ['symfony.com'],
+            ['example.co.uk'],
+            ['example.fr'],
+            ['example.com'],
+            ['xn--diseolatinoamericano-66b.com'],
+            ['xn--ggle-0nda.com'],
+            ['www.xn--simulateur-prt-2kb.fr'],
+            [sprintf('%s.com', str_repeat('a', 20))],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidDomains
+     */
+    public function testInvalidDomainsRaiseViolationIfTldRequired($domain)
```

Comment:
We should also have tests for domains that are invalid when TLDs are required, but that are valid when no TLD is needed.

Reply:
![image](https://user-images.githubusercontent.com/1675033/65621591-b8e74f00-dfcc-11e9-9e73-e8c19c49e469.png)